src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -197,12 +198,35 @@ def __enter__(self):
         # Starting a new process in non-fork mode requires to allocate memory. Determine the maximum number of processes
         # based on available memory to avoid memory bursting.
         if not self._use_fork:
-            available_max_worker_count = get_available_max_worker_count()
-            self._n_process = min(self._worker_count, self._nlines, available_max_worker_count)
-            bulk_logger.info(f"Not using fork, process count: {self._n_process}")
+            estimated_available_worker_count = get_available_max_worker_count()
+            if self._use_default_worker_count:
+                self._n_process = min(self._worker_count, self._nlines, estimated_available_worker_count)
+                bulk_logger.info(
+                    f"Not using fork, the environment variable PF_WORKER_COUNT is not set. Calculate the estimated "
+                    f"available worker count based on the current system available memory and process memory. Take the "
+                    f"minimum value among the calculated value: {estimated_available_worker_count}, "
+                    f"the default value for worker_count: {self._worker_count}, and the row count: "
+                    f"{self._nlines}. Process count: {self._n_process}")
+            else:
+                self._n_process = self._worker_count
+                if estimated_available_worker_count < self._worker_count:
+                    bulk_logger.warning(
+                        f"The estimated available worker count calculated based on the system available memory "
+                        f"is {estimated_available_worker_count}, but the PF_WORKER_COUNT is set to "
+                        f"{self._worker_count}. This may affect optimal memory usage and performance. "
+                        f"Process count: {self._n_process}")
+                else:
+                    log_process_count_info(self._worker_count)
         else:
-            self._n_process = min(self._worker_count, self._nlines)
-            bulk_logger.info(f"Using fork, process count: {self._n_process}")
+            if self._use_default_worker_count:
+                self._n_process = min(self._worker_count, self._nlines)
+                bulk_logger.info(
+                    f"Using fork, the environment variable PF_WORKER_COUNT is not set. Take the "
+                    f"minimum value among the default value for worker_count {self._worker_count} and "
+                    f"the row count: {self._nlines}. Process count: {self._n_process}")
+            else:
+                self._n_process = self._worker_count
+                log_process_count_info(self._worker_count)
```

Comment:
Though extracting a new method `log_process_count_info` reduces some redundant code, it reduces readability as other cases are printing logs in place.

Reply:
removed. fixed